core/search/disjunction.hpp

```diff
@@ -449,6 +683,11 @@ class disjunction : public doc_iterator_base, score_ctx {
     }
 
     doc_.value = lead().value();
+
+    if /*constexpr*/ (std::is_same<Adapter, position_score_iterator_adapter<DocIterator>>::value) {
+      hitch_all_iterators();
+      this->update_attribute_range(heap_, itrs_);
+    }
```

Comment:
let's avoid having this logic as a part of next/seek call, use attribute+visitor instead